sysdig/data_source_sysdig_fargate_workload_agent.go

```diff
@@ -0,0 +1,208 @@
+package sysdig
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/falcosecurity/kilt/runtimes/cloudformation/cfnpatcher"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const agentinoKiltDefinition = `build {
+    entry_point: ["/opt/draios/bin/instrument"]
+    command: ${?original.entry_point} ${?original.command}
+    environment_variables: {
+        "SYSDIG_ORCHESTRATOR": ${config.orchestrator_host}
+        "SYSDIG_ORCHESTRATOR_PORT": ${config.orchestrator_port}
+        "SYSDIG_COLLECTOR": ${config.collector_host}
+        "SYSDIG_COLLECTOR_PORT": ${config.collector_port}
+        "SYSDIG_ACCESS_KEY": ${config.sysdig_access_key}
+        "SYSDIG_LOGGING": ""
+    }
+    mount: [
+        {
+            name: "SysdigInstrumentation"
+            image: ${config.agent_image}
+            volumes: ["/opt/draios"]
+            entry_point: ["/opt/draios/bin/waitforever"]
+        }
+    ]
+}`
+
+func dataSourceSysdigFargateWorkloadAgent() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceSysdigFargateWorkloadAgentRead,
+
+		Schema: map[string]*schema.Schema{
+			"container_definitions": {
+				Type:        schema.TypeString,
+				Description: "the input Fargate container definitions to instrument with the Sysdig workload agent",
+				Required:    true,
+			},
+			"sysdig_access_key": {
+				Type: schema.TypeString,
+				Description: "the Sysdig access key",
+				Required: true,
+			},
+			"workload_agent_image": {
+				Type: schema.TypeString,
+				Description: "the Sysdig workload agent image",
+				Required: true,
+			},
+			"image_auth_secret": {
+				Type: schema.TypeString,
+				Description: "registry authentication secret",
+				Optional: true,
+			},
```

Comment:
this will be unused as our images are currently downloadable without auth

Reply:
I expect this is for people mirroring the image to their own private repo that might need authentication?